rep-0145.rst

```diff
@@ -0,0 +1,190 @@
+REP: 145
+Title: Conventions for IMU Sensor Drivers
+Author: Paul Bovbel <pbovbel@clearpathrobotics.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 02-Feb-2015
+Post-History: 02-Feb-2015
+
+
+Abstract
+========
+
+This REP defines common parameters, topics, namespaces, and data processing conventions for drivers of Inertial Measurement Unit (IMU) sensors. This includes accelerometers, gyroscopes, magnetomers, and any combination thereof.
+
+Motivation
+==========
+
+This REP seeks to standardize an interface for IMU drivers in the ROS ecosystem, providing for IMU-type sensors with a variety of capabilities and conventions. By formalizing a common approach based on existing driver implementations, this REP serves as a guideline for new driver development, while minimizing the changes required to non-conforming implementations. This REP discourages the in-driver manipulation of sensor data, by specifying the conditions where sensor data will be properly handled by ROS transform and coordinate specifications. The REP also specifies fallback approaches to correcting sensor data that does not conform to ROS conventions.
+
+Specification
+=============
+
+Frame Conventions
+-----------------
+
+An IMU device may measure data with respect to two frames, specified by the manufacturer:
+
+* The **body frame** represents the internal device axes. These may be found either in the device specification documents, and sometimes printed directly on the device body. This frame is fixed to the device orientation.
+
+* If an orientation estimate is provided (see `Data Reporting`_), the **world frame** represents the external reference frame for the device. This frame is fixed in one of two orientations, with the relevant conventions from REP 103 [1]_ :
+
+  - For NED type IMUs, the orientation is x-forward, y-right, and z-down. If an absolute yaw reference relative to magnetic north is available via the magnetometer, the orientation is x-north, y-east, z-down.
+
+  - For ENU type IMUs, the orientation is x-forward, y-left, and z-up. If an absolute yaw reference relative to magnetic north is available via the magnetometer, the orientation is x-east, y-north, z-up.
+
+* If the device does not have an absolute yaw reference (magnetometer), the world frame is only aligned with an external reference along the `z` axis. The `x` and `y` axes are aligned relative to the power-on position of the sensor.
```

Comment:
I've tried clearing up the language a little bit, especially since it was getting a little too if..else..
If you have any references I should take a look at for better language, or a better wording you have in mind, please pull away!